src/omero_zarr/__init__.py

```diff
@@ -1,6 +1,6 @@
 from ._version import version as __version__
 
-ngff_version = "0.3"
+ngff_version = "0.4"
```

Comment:
Since you are using the low-level `ome_zarr` writer APIs, there is a possibility of creating a mix of spec versions. I wonder whether this should not become `CurrentFormat().version` to unify?